File: src/main/java/com/ject/studytrip/pomodoro/domain/policy/PomodoroPolicy.java
package com.ject.studytrip.pomodoro.domain.policy;

import com.ject.studytrip.global.exception.CustomException;
import com.ject.studytrip.pomodoro.domain.error.PomodoroErrorCode;
import com.ject.studytrip.pomodoro.domain.model.Pomodoro;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class PomodoroPolicy {
    public static void validateNotDeleted(Pomodoro pomodoro) {
        if (pomodoro.getDeletedAt() != null)
            throw new CustomException(PomodoroErrorCode.POMODORO_ALREADY_DELETED);
    }
}

File: src/main/kotlin/com/ject/studytrip/pomodoro/domain/policy/PomodoroPolicy.kt
package com.ject.studytrip.pomodoro.domain.policy

import com.ject.studytrip.global.exception.CustomException
import com.ject.studytrip.pomodoro.domain.error.PomodoroErrorCode
import com.ject.studytrip.pomodoro.domain.model.Pomodoro

object PomodoroPolicy {
    fun validateNotDeleted(pomodoro: Pomodoro) {
        if (pomodoro.isDeleted) {
            throw CustomException(PomodoroErrorCode.POMODORO_ALREADY_DELETED)
        }
    }

    fun validateTotalFocusTimeNotNegative(totalFocusTime: Int) {
        if (totalFocusTime < 0) {
            throw CustomException(PomodoroErrorCode.POMODORO_NEGATIVE_FOCUS_TIME)
        }
    }
}
